ocp_resources/node_network_configuration_policy.py

```diff
@@ -322,10 +323,37 @@ def _absent_interface(self):
         if self.ports:
             self.add_ports()
 
+        initial_transition_time = self._get_nncp_configured_last_transition_time()
         ResourceEditor(
             patches={self: {"spec": {"desiredState": {"interfaces": self.desired_state["interfaces"]}}}}
         ).update()
 
+        # If the setup NNCP failed - we don't need to verify its teardown status time is updated.
+        if initial_transition_time:
+            self._wait_for_nncp_status_update(initial_transition_time=initial_transition_time)
+
+    def _get_nncp_configured_last_transition_time(self):
+        for condition in self.instance.status.conditions:
+            if (
+                condition["type"] == NodeNetworkConfigurationPolicy.Conditions.Type.AVAILABLE
+                and condition["status"] == "True"
+                and condition["reason"] == NodeNetworkConfigurationPolicy.Conditions.Reason.SUCCESSFULLY_CONFIGURED
+            ):
+                return condition["lastTransitionTime"]
+
+    @retry(
+        wait_timeout=TIMEOUT_1MINUTE,
+        sleep=TIMEOUT_5SEC,
+    )
+    def _wait_for_nncp_status_update(self, initial_transition_time):
+        date_format = "%Y-%m-%dT%H:%M:%SZ"
+        formatted_initial_transition_time = datetime.strptime(initial_transition_time, date_format)
+        return any(
```

Comment:
You know that is only a single Available condition, why not access it directly? Something like:
    conditions = self.instance.get("status", {}).get("conditions", [])
    if conditions and conditions[0]["type"] == "Available":
        last_transition_time = datetime.strptime(conditions[0]["lastTransitionTime"], date_format)
        return last_transition_time > formatted_initial_transition_time

Reply:
You cannot assume that a target member is located in a specific index of a list.
You cannot access `conditions[0]' directly and assume the this is the condition that you are interested in. You must iterate over all the conditions in the list and find the one that you are interested in.